### PR TITLE
Feat search #144

### PR DESCRIPTION
**Pull request by fromdusttilldawn**

### CODE REVIEW

File: helpers/renderingInfoScript.js [hunk 1]
```diff
@@ -217,9 +217,91 @@ function getMinibarsScript(context) {
...
+    }
+
+    function ${filterRows}(filter) {
+      filter = filter.toUpperCase();
+      
+      if (filter.length < 2) return;
+
+      ${dataObject}.tableElement.querySelectorAll('tbody tr .q-table__cell--text').forEach(
+        function(cellElement, index) {
+          txtValue = cellElement.innerText.toUpperCase();
```
Comment (philipkueng) [inline helpers/renderingInfoScript.js:259]: Using a more expressive variable name should help understanding what you're doing here

File: helpers/renderingInfoScript.js [hunk 2]
```diff
@@ -217,9 +217,91 @@ function getMinibarsScript(context) {
...
+        }
+      );
+    }
+
+    function ${searchFormInputAddEventListeners}() {
+      document.querySelector('#search-form-input').addEventListener('input', function(event) {
+        var filter = event.target.value;
+
+        if (filter.length == 0) {
+          // No filter = show only x rows
```
Comment (philipkueng) [inline helpers/renderingInfoScript.js:277]:
```suggestion
          // No filter = show default view with show more button (15 rows)
```

File: helpers/renderingInfoScript.js (hunk 1 above)
Comment (fromdusttilldawn) [reply helpers/renderingInfoScript.js:259]: You're refering to the variable `txtValue` right?

File: views/table.html [hunk 3]
```diff
@@ -3,7 +3,11 @@
   <h3 class="s-q-item__title">{{ item.title }}</h3>{% endif %} {%- if item.subtitle and item.subtitle !== '' %}
   <div class="s-q-item__subtitle">{{ item.subtitle }}</div>
   {%- endif %}
-  <div style="overflow-x: auto">
+  <div style="overflow-x: auto;padding: 0 1px;">
```
Comment (philipkueng) [inline views/table.html:6]: What's the `padding` for? I tried to remove it, haven't seen any difference or uses to it.

Comment (fromdusttilldawn) [reply views/table.html:6]: Yes, exactly. The `padding: 0 1px;` is more of a workaround, than a final solution.

File: views/table.html [hunk 4]
```diff
@@ -3,7 +3,11 @@
...
+    {%- if item.options.showTableSearch === true %}
+    <label for="search-form-input" hidden>Suchen</label>
+    <input id="search-form-input" class="s-input-field" type="search" placeholder="Bitte Suche eingeben" maxlength="20" value="" autocapitalize="off" autocomplete="off" spellcheck="false">
```
Comment (philipkueng) [inline views/table.html:9]: using classes, not ids since there can be multiple tables in one article

Comment (fromdusttilldawn) [reply views/table.html:9]: Didn't think about that. Good point! Should I rather put the `input` element inside the `label` tag like this?
```
<label hidden>Suchen
  <input>
</label>
```

Comment (philipkueng) [reply views/table.html:9]: Sounds good to me

Comment (fromdusttilldawn) [reply views/table.html:9]: I'll use the `aria-label` for this (as discussed)

File: README.md [hunk 5]
```diff
@@ -182,6 +182,21 @@ Footnotes are a feature to display annotations in the table and the sources in t
 
 [to the top](#table-of-contents)
 
+#### Table search
+
+<img src="/doc/table-search.png" align="right" width=300 height=400>
```
Comment (philipkueng) [inline README.md:187]: image is not added to repository

Comment (fromdusttilldawn) [reply README.md:187]: You're too fast 😅 I'm on it

File: README.md [hunk 6]
```diff
@@ -182,6 +182,21 @@ Footnotes are a feature to display annotations in the table and the sources in t
...
+
+This feature makes large tables searchable.
+
+###### Implementation details
+- By default the table is collapsed and the `q-table_show-more-button` is visible at the bottom of the table ([see Collapsable table](#collapsable-table)).
+- When the user starts typing, all rows will be made visible and the `q-table_show-more-button` disappears.
+- The actual search function triggers, as soon as the user types the second character.
+- At the moment only `q-table__cell--text` cells are searched.
```
Comment (philipkueng) [inline README.md:195]:
```suggestion
- The filter searches through text-based columns only
```

File: views/table.html [hunk 7]
```diff
@@ -3,7 +3,12 @@
   <h3 class="s-q-item__title">{{ item.title }}</h3>{% endif %} {%- if item.subtitle and item.subtitle !== '' %}
   <div class="s-q-item__subtitle">{{ item.subtitle }}</div>
   {%- endif %}
-  <div style="overflow-x: auto">
+  <div style="overflow-x: auto;">
+    {%- if noInteraction !== true and item.options.showTableSearch === true %}
+    <div class="search-form" style="padding: 0 1px">
```
Comment (philipkueng) [inline views/table.html:8]: why class and style?

Comment (fromdusttilldawn) [reply views/table.html:8]: Lazyness. Will fix and also change the naming

**Review by philipkueng — APPROVED** [review]
Good job! Congratulations on your very first feature.

**Pull request merged**